azure-mgmt-trafficmanager/src/main/java/com/microsoft/azure/management/trafficmanager/implementation/TrafficManagerNestedProfileEndpointImpl.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.trafficmanager.implementation;
+
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.trafficmanager.TrafficManagerNestedProfileEndpoint;
+
+/**
+ * Implementation for {@link TrafficManagerNestedProfileEndpoint}.
+ */
+class TrafficManagerNestedProfileEndpointImpl extends TrafficManagerEndpointImpl
+    implements TrafficManagerNestedProfileEndpoint {
+    TrafficManagerNestedProfileEndpointImpl(String name,
+                                            TrafficManagerProfileImpl parent,
+                                            EndpointInner inner,
+                                            EndpointsInner client) {
+        super(name, parent, inner, client);
+    }
+
+    @Override
+    public String nestedProfileId() {
+        return inner().targetResourceId();
+    }
+
+    @Override
+    public int minChildEndpoints() {
```

Comment:
the inner naming is bad... It uses an abbreviation, and its plural nouns suggests it returns a collection. We should expose it with better naming: how about minimumChildEndpointCount()?